[Logs / Metrics UI] Convert logs and metrics pages to the new Observability page template

Kerry350 opened the pull request:
## Summary

This PR takes the work in #98212 and extends it to all Logs and Metrics pages.

Closes #99966
Closes #99968
Closes #99970
Closes #99971

## Examples

(These are collapsed as there's a few. Please also note that there are many variations of state within each page, e.g. loading states, source configuration loading / error states, lack of privileges / lack of indices states etc, these aren't captured here.)

<details>
<summary>Screenshots</summary>

![Screenshot 2021-06-03 at 17 35 30](https://user-images.githubusercontent.com/471693/120681868-bfadb100-c493-11eb-9de4-e3c58a3cb65b.png)

![Screenshot 2021-06-03 at 17 35 36](https://user-images.githubusercontent.com/471693/120681882-c2a8a180-c493-11eb-8f57-95c01292b1e5.png)

![Screenshot 2021-06-03 at 17 35 42](https://user-images.githubusercontent.com/471693/120681883-c3413800-c493-11eb-9bae-eccbf065c05a.png)

![Screenshot 2021-06-03 at 17 42 21](https://user-images.githubusercontent.com/471693/120681886-c3d9ce80-c493-11eb-9aba-2b61a175edc3.png)

![Screenshot 2021-06-03 at 17 42 30](https://user-images.githubusercontent.com/471693/120681887-c3d9ce80-c493-11eb-8ff2-d1e5afd95ef5.png)

![Screenshot 2021-06-03 at 17 42 44](https://user-images.githubusercontent.com/471693/120681890-c4726500-c493-11eb-8c8c-16ebf08864b3.png)

![Screenshot 2021-06-04 at 14 07 32](https://user-images.githubusercontent.com/471693/120805993-3f905580-c53e-11eb-948e-f77af3a681c2.png)

</details>

## Known issues

There is a scrollbar "jitter" on the Metrics inventory page when loading, however this is present in `master` so I'd rather divert this as the inventory page has already taken some time to fix. 

![jitter](https://user-images.githubusercontent.com/471693/120682345-45c9f780-c494-11eb-8f33-f1f73de081e1.gif)

## Follow-up tasks

- clean up of remove the `Error` page in `public/pages/error.tsx`
- The inner content of the Inventory page needs reorganising / refactoring. E.g. the load view button, moving the open history button, and moving the scale to the right.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128812/)
* Commit: e04d4af90e44d65e128d78f20632b794ef9bf89d
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128812/execution/node/423/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128812/","number":"128812","commit":"e04d4af90e44d65e128d78f20632b794ef9bf89d"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128989/)
* Commit: f386eea73434d5b19abd7706db9ac4e332acbf98
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128989/execution/node/413/log)

### History
* :broken_heart: [Build #128812](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128812/) failed e04d4af90e44d65e128d78f20632b794ef9bf89d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128812/","number":"128812","commit":"e04d4af90e44d65e128d78f20632b794ef9bf89d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128989/","number":"128989","commit":"f386eea73434d5b19abd7706db9ac4e332acbf98"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/logs-metrics-ui (Team:logs-metrics-ui)

Kerry350 commented:
(There's an integration test that still needs fixing, but I'm marking this open as the review is already in progress)

Kerry350 commented:
Just noticed this is broken in the metrics flyout context. Will fix. 

![Screenshot 2021-06-04 at 15 11 38](https://user-images.githubusercontent.com/471693/120814893-3b1c6a80-c547-11eb-920c-72247648df71.png)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/)
* Commit: fd2826afdc5d195f2da21ff8bf1b8fcb69f1a47d
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101239/fd2826afdc5d195f2da21ff8bf1b8fcb69f1a47d)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/execution/node/713/log)
* [Execute xpack-kibana-ciGroup3](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/execution/node/628/log)

### History
* :broken_heart: [Build #128989](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128989/) failed f386eea73434d5b19abd7706db9ac4e332acbf98
* :broken_heart: [Build #128812](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128812/) failed e04d4af90e44d65e128d78f20632b794ef9bf89d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @Kerry350

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128812/","number":"128812","commit":"e04d4af90e44d65e128d78f20632b794ef9bf89d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128989/","number":"128989","commit":"f386eea73434d5b19abd7706db9ac4e332acbf98"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/","number":"129170","commit":"fd2826afdc5d195f2da21ff8bf1b8fcb69f1a47d"}]}
PIPELINE-->

weltenwort commented on x-pack/plugins/infra/public/plugin.ts:
```diff
@@ -50,6 +51,29 @@ export class Plugin implements InfraClientPluginClass {
 
+    /** !! Need to be kept in sync with the deepLinks in x-pack/plugins/infra/public/plugin.ts */
+    pluginsSetup.observability.navigation.registerSections(
+      of([
```
This unconditionally registers the apps regardless of the space and role settings. We probably need to check the `show` capabilities as in

![image](https://user-images.githubusercontent.com/973741/120819369-cc91d980-c553-11eb-94ba-0e55fd4f025b.png)

Kerry350 replied:
Oh wow, yep 🙈 

Of course I would remember to add the comment back left by the platform team, but forget the capability check.

Also related: I think we should make the Observability plugin required. It's currently optional. I would either need to move this registration into our `if`, or we make it required. Now that navigation is routed through the Observability plugin I think it's fair to require it.

weltenwort replied:
> I think we should make the Observability plugin required.

Agreed. The impact on the user should be minimal since there is no separate permission for the observability plugin to take away and disabling plugins via the `yaml` is probably going to go away soon.

weltenwort reviewed (changes requested):
After just looking at the Logs UI I really like how many special cases you removed :clap:

So far I only noticed one major problem concerning the lack capability checks for the navigation section registration.

Aside from that only a few small questions arose. I'll look at the changes for the Metrics UI next.

Kerry350 commented on x-pack/plugins/infra/public/components/error_page.tsx:
```diff
@@ -5,72 +5,59 @@
+        navigation: { PageTemplate },
+      },
+    },
+  } = useKibanaContextForPlugin();
```
Yeah, this is a good point. I do think it would be worth it.

>  And would we want to have separate ones for logs and metrics "just in case"?

Yeah 👍 

At this point in time I prefer to separate the two where we can.

Kerry350 commented on x-pack/plugins/infra/public/pages/logs/log_entry_categories/page_content.tsx:
```diff
@@ -57,24 +79,52 @@ export const LogEntryCategoriesPageContent = () => {
+const CategoriesPageTemplate: React.FC<{
+  PageTemplate: React.ComponentType<LazyObservabilityPageTemplateProps>;
+  isEmptyState?: boolean;
+}> = ({ PageTemplate, isEmptyState = false, children }) => {
```
It could. I'd originally played around with a few variations here, this was unnecessarily left behind.

Kerry350 commented on x-pack/plugins/infra/public/pages/logs/log_entry_categories/page_content.tsx:
```diff
@@ -23,8 +23,22 @@ import { useLogAnalysisCapabilitiesContext } from '../../../containers/logs/log_
+import { useKibanaContextForPlugin } from '../../../hooks/use_kibana';
+import type { LazyObservabilityPageTemplateProps } from '../../../../../observability/public';
+
+const logCategoriesTabTitle = i18n.translate('xpack.infra.logs.logCategoriesTitle', {
```
Fair point 🙈

Kerry350 commented on x-pack/plugins/infra/public/pages/logs/log_entry_rate/page_content.tsx:
```diff
@@ -25,10 +25,24 @@ import { useLogEntryCategoriesModuleContext } from '../../../containers/logs/log
 
 const JOB_STATUS_POLLING_INTERVAL = 30000;
 
+const anomaliesTitle = i18n.translate('xpack.infra.logs.anomaliesPageTitle', {
```
Yeah, 100%. I'd pondered exactly this and some sort of consistent constants file for us.

Kerry350 commented on x-pack/plugins/infra/public/pages/logs/log_entry_categories/page_content.tsx:
```diff
@@ -57,24 +79,52 @@ export const LogEntryCategoriesPageContent = () => {
   } else {
     return (
-      <>
+      <CategoriesPageTemplate PageTemplate={PageTemplate} isEmptyState={true}>
```
Good spot 😬

Kerry350 replied:
Interestingly I couldn't replicate the behaviour 🤔  (or maybe I misunderstood the manifestation) 

![flyout](https://user-images.githubusercontent.com/471693/121394365-2d048a80-c949-11eb-8609-39f6f32a6c1c.gif)

I've changed the layout of the components regardless though.

weltenwort replied:
I probably explained it incorrectly. It happens when you go through the process of creating the ML jobs. The whole page template will re-render a different branch of the large conditional in the page component because the jobs now exist, which causes the fly-out to be re-mounted in a different hierarchy location and trigger the slide-in animation several times.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/)
* Commit: 0183e9efe7042cc40ab89cc80334f31f234c2b30
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101239/0183e9efe7042cc40ab89cc80334f31f234c2b30)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/execution/node/675/log)
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/execution/node/961/log)

### History
* :broken_heart: [Build #129170](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/) failed fd2826afdc5d195f2da21ff8bf1b8fcb69f1a47d
* :broken_heart: [Build #128989](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128989/) failed f386eea73434d5b19abd7706db9ac4e332acbf98
* :broken_heart: [Build #128812](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128812/) failed e04d4af90e44d65e128d78f20632b794ef9bf89d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @Kerry350

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128812/","number":"128812","commit":"e04d4af90e44d65e128d78f20632b794ef9bf89d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128989/","number":"128989","commit":"f386eea73434d5b19abd7706db9ac4e332acbf98"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/","number":"129170","commit":"fd2826afdc5d195f2da21ff8bf1b8fcb69f1a47d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/","number":"129195","commit":"0183e9efe7042cc40ab89cc80334f31f234c2b30"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129245/)
* Commit: 826a3d91cd95e5d45ae3bb33ccfff87ff80e7424
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101239/826a3d91cd95e5d45ae3bb33ccfff87ff80e7424)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/71b4c38c4a579b0b4871b9f437d6d855f5076511) | [after](https://github.com/elastic/kibana/commit/826a3d91cd95e5d45ae3bb33ccfff87ff80e7424) | diff |
| --- | --- | --- | --- |
| `infra` | 967 | 959 | -8 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/71b4c38c4a579b0b4871b9f437d6d855f5076511) | [after](https://github.com/elastic/kibana/commit/826a3d91cd95e5d45ae3bb33ccfff87ff80e7424) | diff |
| --- | --- | --- | --- |
| `observability` | 192 | 193 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/71b4c38c4a579b0b4871b9f437d6d855f5076511) | [after](https://github.com/elastic/kibana/commit/826a3d91cd95e5d45ae3bb33ccfff87ff80e7424) | diff |
| --- | --- | --- | --- |
| `infra` | 1.8MB | 1.8MB | -8.7KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/71b4c38c4a579b0b4871b9f437d6d855f5076511) | [after](https://github.com/elastic/kibana/commit/826a3d91cd95e5d45ae3bb33ccfff87ff80e7424) | diff |
| --- | --- | --- | --- |
| `infra` | 140.6KB | 141.1KB | +486.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/71b4c38c4a579b0b4871b9f437d6d855f5076511) | [after](https://github.com/elastic/kibana/commit/826a3d91cd95e5d45ae3bb33ccfff87ff80e7424) | diff |
| --- | --- | --- | --- |
| `observability` | 192 | 193 | +1 |


</details>

### History
* :broken_heart: [Build #129195](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/) failed 0183e9efe7042cc40ab89cc80334f31f234c2b30
* :broken_heart: [Build #129170](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/) failed fd2826afdc5d195f2da21ff8bf1b8fcb69f1a47d
* :broken_heart: [Build #128989](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128989/) failed f386eea73434d5b19abd7706db9ac4e332acbf98
* :broken_heart: [Build #128812](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128812/) failed e04d4af90e44d65e128d78f20632b794ef9bf89d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @Kerry350

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128812/","number":"128812","commit":"e04d4af90e44d65e128d78f20632b794ef9bf89d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128989/","number":"128989","commit":"f386eea73434d5b19abd7706db9ac4e332acbf98"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/","number":"129170","commit":"fd2826afdc5d195f2da21ff8bf1b8fcb69f1a47d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/","number":"129195","commit":"0183e9efe7042cc40ab89cc80334f31f234c2b30"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129245/","number":"129245","commit":"826a3d91cd95e5d45ae3bb33ccfff87ff80e7424"}]}
PIPELINE-->

formgeist commented on x-pack/plugins/infra/public/pages/logs/stream/page_logs_content.tsx:
```diff
@@ -206,6 +207,7 @@ export const LogsPageLogsContent: React.FunctionComponent = () => {
       <WithLogTextviewUrlState />
       <WithFlyoutOptionsUrlState />
       <LogsToolbar />
+      <EuiSpacer size="l" />
```
I personally don't think we need the additional borders added to the stream. The typical table-like components don't have similar effects.

formgeist replied:
```suggestion
      <EuiSpacer size="m" />
```

I'd also not make it any taller than size `m`

weltenwort reviewed (commented):
The metrics-related part looks good as well :clap: As discussed on another channel we should ask the metrics ui team to double-check and follow up any quirks that result from the new template's paddings, borders and such.

Kerry350 commented on x-pack/plugins/infra/public/pages/metrics/metric_detail/components/side_nav.tsx:
```diff
@@ -32,22 +31,23 @@ export const MetricsSideNav = ({ loading, name, items }: Props) => {
+const StickyPageSideBar = euiStyled(EuiPageSideBar)`
+  position: sticky;
+  top: ${(props) =>
+    parseFloat(props.theme.eui.euiHeaderHeightCompensation) * 2 +
```
I will try and investigate this a bit more. I don't like magic numbers but was struggling to find the "origin" for this one. There's lots of places doing this * 2 in Kibana to account for the header fixed space, but I struggled to find why there wasn't a separate variable for that.

cchaos replied:
It's because there are **two** headers 😄 . But more importantly, why are these extra styles necessary? Is the template not behaving correctly? This should all be accounted for.

Kerry350 replied:
@cchaos Oh, duh 😅, thanks! 

This is a use case on the node details page in Metrics. There's a loose design here: https://github.com/elastic/kibana/issues/99970.

The additional side navigation with the metrics uses a fixed position. 

![fixed-nav](https://user-images.githubusercontent.com/471693/121216221-9b2d4c80-c878-11eb-8b55-dfe43f5d35f6.gif)

Would there be a better way to do this? Or do you recommend removing the fixed nature?

cchaos replied:
OH it's a secondary side-nav, not the main Solution nav. I see. Does applying `position="sticky"` to the EuiPageSideBar component not work? That should then ensure that the top position is always correct because of [this mixin](https://github.com/elastic/kibana/blob/356b7d6c9202269d00b70886f95bfe0eb97555c5/src/core/public/rendering/_base.scss#L40-L59).

Kerry350 replied:
Ah - I somehow missed the `sticky` prop on the `<EuiPageSideBar />`. That works 🙌 

Thanks for the input.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129812/)
* Commit: 17605521713075a28e9318546946b34b32cc11eb
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101239/17605521713075a28e9318546946b34b32cc11eb)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/71b4c38c4a579b0b4871b9f437d6d855f5076511) | [after](https://github.com/elastic/kibana/commit/17605521713075a28e9318546946b34b32cc11eb) | diff |
| --- | --- | --- | --- |
| `infra` | 967 | 959 | -8 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/71b4c38c4a579b0b4871b9f437d6d855f5076511) | [after](https://github.com/elastic/kibana/commit/17605521713075a28e9318546946b34b32cc11eb) | diff |
| --- | --- | --- | --- |
| `observability` | 192 | 193 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/71b4c38c4a579b0b4871b9f437d6d855f5076511) | [after](https://github.com/elastic/kibana/commit/17605521713075a28e9318546946b34b32cc11eb) | diff |
| --- | --- | --- | --- |
| `infra` | 1.8MB | 1.8MB | -8.7KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/71b4c38c4a579b0b4871b9f437d6d855f5076511) | [after](https://github.com/elastic/kibana/commit/17605521713075a28e9318546946b34b32cc11eb) | diff |
| --- | --- | --- | --- |
| `infra` | 140.6KB | 141.1KB | +486.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/71b4c38c4a579b0b4871b9f437d6d855f5076511) | [after](https://github.com/elastic/kibana/commit/17605521713075a28e9318546946b34b32cc11eb) | diff |
| --- | --- | --- | --- |
| `observability` | 192 | 193 | +1 |


</details>

### History
* :green_heart: [Build #129245](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129245/) succeeded 826a3d91cd95e5d45ae3bb33ccfff87ff80e7424
* :broken_heart: [Build #129195](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/) failed 0183e9efe7042cc40ab89cc80334f31f234c2b30
* :broken_heart: [Build #129170](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/) failed fd2826afdc5d195f2da21ff8bf1b8fcb69f1a47d
* :broken_heart: [Build #128989](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128989/) failed f386eea73434d5b19abd7706db9ac4e332acbf98
* :broken_heart: [Build #128812](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128812/) failed e04d4af90e44d65e128d78f20632b794ef9bf89d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @Kerry350

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128812/","number":"128812","commit":"e04d4af90e44d65e128d78f20632b794ef9bf89d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128989/","number":"128989","commit":"f386eea73434d5b19abd7706db9ac4e332acbf98"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/","number":"129170","commit":"fd2826afdc5d195f2da21ff8bf1b8fcb69f1a47d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/","number":"129195","commit":"0183e9efe7042cc40ab89cc80334f31f234c2b30"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129245/","number":"129245","commit":"826a3d91cd95e5d45ae3bb33ccfff87ff80e7424"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129812/","number":"129812","commit":"17605521713075a28e9318546946b34b32cc11eb"}]}
PIPELINE-->

Kerry350 commented on x-pack/plugins/infra/public/apps/metrics_app.tsx:
```diff
@@ -29,7 +28,7 @@ export const renderApp = (
   const storage = new Storage(window.localStorage);
 
-  prepareMountElement(element);
+  prepareMountElement(element, 'infraMetricsPage');
```
This benefits the functional tests. We previously had a wrapping element in the hierarchy ( a `<ColumnarPage />` I think that added these, but I've removed those. I added it here so it belongs to the top most elements we own.

These get used in the functional tests as a pretty generic "logs / metrics loaded". 

The tests could be reworked to care about the specific page, rather than a more generic wrapper. And then the specific page template strings could be used instead.

This really just pertains to the feature controls / spaces tests. Let me know if you think it's worth the time here.

weltenwort replied:
Thanks for the explanation. I think it's fine for now. :+1:

Kerry350 commented:
@weltenwort 

This should be good to go now (well, the tests are still running locally but changes should be good).

Regarding https://github.com/elastic/kibana/pull/101239#issuecomment-854758976 I ended up changing the subscription splash content contents (for the ML pages / flyout). Now that we've moved over to the consistency of the `<EuiEmptyPrompts />` for loading / error / missing privileges states the content there felt out of place.

The footer doesn't really fit into the `<EuiEmptyPrompt />` layout, **but** it transpires that footer link was navigating to a page that 404s anyway. So I've just removed it.

Rather than try to accommodate the custom image I used the icon type that was closest to the image.

It looks like this:

![Screenshot 2021-06-09 at 16 55 39](https://user-images.githubusercontent.com/471693/121395320-33dfcd00-c94a-11eb-9abc-230528865307.png)

![Screenshot 2021-06-09 at 16 57 05](https://user-images.githubusercontent.com/471693/121395321-34786380-c94a-11eb-989e-7575bbaf295a.png)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130219/)
* Commit: f2738b15db010ae090f6af6c4bf9be7af1c5f268
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130219/execution/node/430/log)

### History
* :green_heart: [Build #129812](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129812/) succeeded 17605521713075a28e9318546946b34b32cc11eb
* :green_heart: [Build #129245](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129245/) succeeded 826a3d91cd95e5d45ae3bb33ccfff87ff80e7424
* :broken_heart: [Build #129195](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/) failed 0183e9efe7042cc40ab89cc80334f31f234c2b30
* :broken_heart: [Build #129170](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/) failed fd2826afdc5d195f2da21ff8bf1b8fcb69f1a47d
* :broken_heart: [Build #128989](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128989/) failed f386eea73434d5b19abd7706db9ac4e332acbf98

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @Kerry350

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128989/","number":"128989","commit":"f386eea73434d5b19abd7706db9ac4e332acbf98"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/","number":"129170","commit":"fd2826afdc5d195f2da21ff8bf1b8fcb69f1a47d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/","number":"129195","commit":"0183e9efe7042cc40ab89cc80334f31f234c2b30"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129245/","number":"129245","commit":"826a3d91cd95e5d45ae3bb33ccfff87ff80e7424"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129812/","number":"129812","commit":"17605521713075a28e9318546946b34b32cc11eb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130219/","number":"130219","commit":"f2738b15db010ae090f6af6c4bf9be7af1c5f268"}]}
PIPELINE-->

weltenwort commented:
> The footer doesn't really fit into the `<EuiEmptyPrompt />` layout, **but** it transpires that footer link was navigating to a page that 404s anyway. So I've just removed it.
> 
> Rather than try to accommodate the custom image I used the icon type that was closest to the image.

Fine with me. Hope the authors of the splash screen don't mind :wink:

I'll do another review pass today.

formgeist commented:
@Kerry350 I'm just reviewing the UX (since @katefarrar is away on PTO) and I found a few small visual nits and possible enhancements because we've changed the page layout. Wonder if you have time to address, or we should create follow-up issues?

### Inventory page has custom padding properties that we can probably remove or align

It would be great to vertically align the inner contents of the page. I noticed there's a number of custom padding properties on some styled-components, so perhaps there's an opportunity to reduce the use of those in favor of e.g. EuiPanel.

<details>
<summary>View screens</summary>

<img width="1499" alt="Screenshot 2021-06-10 at 12 16 20" src="https://user-images.githubusercontent.com/4104278/121510003-7b686680-c9e7-11eb-89b3-6710644bbe1c.png">

<img width="1489" alt="Screenshot 2021-06-10 at 12 16 33" src="https://user-images.githubusercontent.com/4104278/121510016-7dcac080-c9e7-11eb-9267-c229f36d6324.png">

</details>

### The date picker styles deviate from the original EUI component

Looks like there are some styles missing from the component, perhaps overwritten?

<img width="544" alt="Screenshot 2021-06-10 at 12 01 31" src="https://user-images.githubusercontent.com/4104278/121510086-920ebd80-c9e7-11eb-950f-ad9bd5278a1b.png">

<img width="469" alt="Screenshot 2021-06-10 at 12 01 37" src="https://user-images.githubusercontent.com/4104278/121510093-94711780-c9e7-11eb-826d-647e8ea877ff.png">

Happy to open follow-up issues if this is a larger refactor of components.

Kerry350 commented:
@formgeist Thanks for doing a UX pass.

> Inventory page has custom padding properties that we can probably remove or align

For this one I'll add it to the Metrics followup ticket which will also be addressing the enhancements (those mentioned in the description under followups). It'll all go hand in hand so makes sense to address it as part of those layout changes. 

> The date picker styles deviate from the original EUI component

I've pushed https://github.com/elastic/kibana/pull/101239/commits/4e72e644be00dfdbc2160fb18e4f9732a6a5cca5 to address this one.

![Screenshot 2021-06-10 at 15 58 54](https://user-images.githubusercontent.com/471693/121548260-c8a70100-ca04-11eb-975f-f4ba2dc81294.png)

(Also just a general note, our experience has degraded in a number of places in terms of the responsive layouts. FlexGroups and FlexItems apply a number of styles at various breakpoints (when `responsive=true`), but because of the page layout and the space taken by the side navigation we ideally need these breakpoints to kick in earlier. But this would really need to be a change at the EUI / wider Kibana level.)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130507/)
* Commit: 340009a71d7e6e954646ff9a27231ea9e9bd9411
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101239/340009a71d7e6e954646ff9a27231ea9e9bd9411)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/ef8eb321cf737584c66a7e6cad9cf333e9fba587) | [after](https://github.com/elastic/kibana/commit/340009a71d7e6e954646ff9a27231ea9e9bd9411) | diff |
| --- | --- | --- | --- |
| `infra` | 967 | 962 | -5 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/ef8eb321cf737584c66a7e6cad9cf333e9fba587) | [after](https://github.com/elastic/kibana/commit/340009a71d7e6e954646ff9a27231ea9e9bd9411) | diff |
| --- | --- | --- | --- |
| `observability` | 194 | 195 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ef8eb321cf737584c66a7e6cad9cf333e9fba587) | [after](https://github.com/elastic/kibana/commit/340009a71d7e6e954646ff9a27231ea9e9bd9411) | diff |
| --- | --- | --- | --- |
| `infra` | 1.8MB | 1.8MB | -10.9KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/ef8eb321cf737584c66a7e6cad9cf333e9fba587) | [after](https://github.com/elastic/kibana/commit/340009a71d7e6e954646ff9a27231ea9e9bd9411) | diff |
| --- | --- | --- | --- |
| `infra` | 140.6KB | 141.3KB | +720.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/ef8eb321cf737584c66a7e6cad9cf333e9fba587) | [after](https://github.com/elastic/kibana/commit/340009a71d7e6e954646ff9a27231ea9e9bd9411) | diff |
| --- | --- | --- | --- |
| `observability` | 194 | 195 | +1 |


</details>

### History
* :broken_heart: [Build #130219](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130219/) failed f2738b15db010ae090f6af6c4bf9be7af1c5f268
* :green_heart: [Build #129812](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129812/) succeeded 17605521713075a28e9318546946b34b32cc11eb
* :green_heart: [Build #129245](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129245/) succeeded 826a3d91cd95e5d45ae3bb33ccfff87ff80e7424
* :broken_heart: [Build #129195](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/) failed 0183e9efe7042cc40ab89cc80334f31f234c2b30
* :broken_heart: [Build #129170](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/) failed fd2826afdc5d195f2da21ff8bf1b8fcb69f1a47d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @Kerry350

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129170/","number":"129170","commit":"fd2826afdc5d195f2da21ff8bf1b8fcb69f1a47d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/","number":"129195","commit":"0183e9efe7042cc40ab89cc80334f31f234c2b30"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129245/","number":"129245","commit":"826a3d91cd95e5d45ae3bb33ccfff87ff80e7424"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129812/","number":"129812","commit":"17605521713075a28e9318546946b34b32cc11eb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130219/","number":"130219","commit":"f2738b15db010ae090f6af6c4bf9be7af1c5f268"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130507/","number":"130507","commit":"340009a71d7e6e954646ff9a27231ea9e9bd9411"}]}
PIPELINE-->

weltenwort reviewed (approved):
Seems to work great! Thanks for putting in all the effort. As always, this looked far easier in planning than it turned out to be.

Kerry350 commented on x-pack/plugins/infra/public/components/subscription_splash_content.tsx:
```diff
@@ -7,25 +7,21 @@
+});
+
+export const SubscriptionSplashContent: React.FC<{
+  includePageTemplate?: boolean;
```
Yeah, I'll split this out, so that it works the same way as the loading components.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131002/)
* Commit: f7df47c5bc53b9738457f22cda7dcd6cc37c395e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101239/f7df47c5bc53b9738457f22cda7dcd6cc37c395e)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/e0a1a34fbc5a92cae76638514779a89cad853fa1) | [after](https://github.com/elastic/kibana/commit/f7df47c5bc53b9738457f22cda7dcd6cc37c395e) | diff |
| --- | --- | --- | --- |
| `infra` | 907 | 902 | -5 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/e0a1a34fbc5a92cae76638514779a89cad853fa1) | [after](https://github.com/elastic/kibana/commit/f7df47c5bc53b9738457f22cda7dcd6cc37c395e) | diff |
| --- | --- | --- | --- |
| `observability` | 195 | 196 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/e0a1a34fbc5a92cae76638514779a89cad853fa1) | [after](https://github.com/elastic/kibana/commit/f7df47c5bc53b9738457f22cda7dcd6cc37c395e) | diff |
| --- | --- | --- | --- |
| `infra` | 1.8MB | 1.7MB | -10.5KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/e0a1a34fbc5a92cae76638514779a89cad853fa1) | [after](https://github.com/elastic/kibana/commit/f7df47c5bc53b9738457f22cda7dcd6cc37c395e) | diff |
| --- | --- | --- | --- |
| `infra` | 140.6KB | 141.3KB | +720.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/e0a1a34fbc5a92cae76638514779a89cad853fa1) | [after](https://github.com/elastic/kibana/commit/f7df47c5bc53b9738457f22cda7dcd6cc37c395e) | diff |
| --- | --- | --- | --- |
| `observability` | 195 | 196 | +1 |


</details>

### History
* :green_heart: [Build #130507](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130507/) succeeded 340009a71d7e6e954646ff9a27231ea9e9bd9411
* :broken_heart: [Build #130219](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130219/) failed f2738b15db010ae090f6af6c4bf9be7af1c5f268
* :green_heart: [Build #129812](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129812/) succeeded 17605521713075a28e9318546946b34b32cc11eb
* :green_heart: [Build #129245](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129245/) succeeded 826a3d91cd95e5d45ae3bb33ccfff87ff80e7424
* :broken_heart: [Build #129195](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/) failed 0183e9efe7042cc40ab89cc80334f31f234c2b30

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @Kerry350

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129195/","number":"129195","commit":"0183e9efe7042cc40ab89cc80334f31f234c2b30"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129245/","number":"129245","commit":"826a3d91cd95e5d45ae3bb33ccfff87ff80e7424"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129812/","number":"129812","commit":"17605521713075a28e9318546946b34b32cc11eb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130219/","number":"130219","commit":"f2738b15db010ae090f6af6c4bf9be7af1c5f268"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130507/","number":"130507","commit":"340009a71d7e6e954646ff9a27231ea9e9bd9411"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131002/","number":"131002","commit":"f7df47c5bc53b9738457f22cda7dcd6cc37c395e"}]}
PIPELINE-->

kibanamachine commented:
## 💚 Backport successful

| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [7.x](https://github.com/elastic/kibana/pull/102112)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/102112">](https://github.com/elastic/kibana/pull/102112) |

This backport PR will be merged automatically after passing CI.
